overlord/snapstate/snapstate.go

```diff
@@ -61,6 +61,10 @@ func needsMaybeCore(typ snap.Type) int {
 }
 
 func doInstall(st *state.State, snapst *SnapState, snapsup *SnapSetup, flags int) (*state.TaskSet, error) {
+	if snapst.IsInstalled() && !snapst.Active {
+		return nil, fmt.Errorf("internal error: cannot update a disabled snap")
```

Comment:
There's no reason for this to be an internal error, as this is a real situation that can happen by the user manipulating the snap state in sane ways. We just want to make sure that case is forbidden here.